packages/go_router/lib/src/router.dart

```diff
@@ -29,7 +30,9 @@ import 'typedefs.dart';
 ///
 /// The `redirect` does top-level redirection before the URIs are parsed by
 /// the `routes`. Consider using [GoRoute.redirect] for individual route
-/// redirection.
+/// redirection. If [BuildContext.dependOnInheritedWidgetOfExactType] is used
+/// during the redirection (which is how `of` method is usually implemented),
+/// a refresh will be triggered if the [InheritedWidget] changes.
```

Comment:
Would "reevaluation" be a better term than refresh here? Refresh makes me think the `GoRoute` will be rebuilt even if the redirect no-ops. I don't feel strongly about this though

```suggestion
/// a reevaluation will be triggered when the [InheritedWidget] changes.
```